New settings to make Held-Suarez 3x faster than before

bischtob opened the pull request:
# Description

This updates the Held-Suarez experiment file with some modified settings that results in a stable simulation (tested to 365 days) with a 3x speedup in terms of WCT compared to the previous settings with the same resolution.

<!--- Please fill out the following section --->

I have

- [x] Written and run all necessary tests with CLIMA by including `tests/runtests.jl`
- [x] Followed all necessary [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and run `julia .dev/format.jl`

<!--- Please leave the following section --->

# For review by CLIMA developers

- [x] There are no open pull requests for this already
- [x] CLIMA developers with relevant expertise have been assigned to review this submission
- [x] The code conforms to the [style guidelines](https://climate-machine.github.io/CLIMA/latest/CodingConventions.html) and has consistent naming conventions. `julia .dev/format.jl` has been run in a separate commit.
- [x] This code does what it is technically intended to do (all numerics make sense physically and/or computationally)

bischtob commented:
@charleskawczynski, any idea what the formatter wants?

charleskawczynski commented:
Run `julia .dev/format.jl`

glwagner commented:
@bischtob do you mind providing a few more details about the settings that were changed to obtain the speed up?

akshaysridhar commented:
bors r+

akshaysridhar commented:
bors r-

akshaysridhar commented:
> Run `julia .dev/format.jl`

With the formatter githooks now there is a prompt which indicates exactly which files need formatting; happy to merge this once the formatter is applied.

jkozdon commented:
```
julia .dev/climaformat.jl experiments/AtmosGCM/heldsuarez.jl 
```

bischtob commented:
Roger. Will do.

bischtob commented:
> @bischtob do you mind providing a few more details about the settings that were changed to obtain the speed up?

I added a stronger sponge layer at the top of the domain to absorb wavy modes, switched to using a reference profile for IMEX that has a continuous derivative, and increased the exponential filtering coefficient (damping).

There is still a lot of speedup potential in adding SGS models to handle accumulation enar the grid scale. The sponge should also ideally be replaced by a "viscous sponge" that increases in strength with height. These two are for another PR.

bischtob commented:
> Run `julia .dev/format.jl`

@charleskawczynski , done.

fxgiraldo commented:
> I added a stronger sponge layer at the top of the domain to absorb wavy modes, switched to using a reference profile for IMEX that has a continuous derivative, and increased the exponential filtering coefficient (damping).
> 
> There is still a lot of speedup potential in adding SGS models to handle accumulation enar the grid scale. The sponge should also ideally be replaced by a "viscous sponge" that increases in strength with height. These two are for another PR.

@bischtob great job! Glad to see that switching the reference profile helped.  Perhaps thinking about how to modify the reference field as suggested by Oswald might be a good idea to test.

tapios commented:
> > I added a stronger sponge layer at the top of the domain to absorb wavy modes, switched to using a reference profile for IMEX that has a continuous derivative, and increased the exponential filtering coefficient (damping).
> > There is still a lot of speedup potential in adding SGS models to handle accumulation enar the grid scale. The sponge should also ideally be replaced by a "viscous sponge" that increases in strength with height. These two are for another PR.
> 
> @bischtob great job! Glad to see that switching the reference profile helped. Perhaps thinking about how to modify the reference field as suggested by Oswald might be a good idea to test.

@fxgiraldo See the atmosphere design doc. There is a smooth analytical reference state in there that should work well (no numerical integration required).

bischtob commented on experiments/AtmosGCM/heldsuarez.jl:
```diff
@@ -22,60 +22,46 @@ const clima_dir = dirname(pathof(CLIMA))
-    p_sfc::FT = MSLP(param_set)
-    T_init::FT = 255
-    T_ref::FT = 300
+    T_ref::FT = 255
```
Unit or init :)? It's just that those could, in princinple, have different values.

charleskawczynski replied:
init :) . If that's the case, then perhaps it would be good to give them different names.

bischtob replied:
K. Let me make some changes.

bischtob replied:
Done.

bischtob commented:
bors r+

kpamnany commented:
To reprise the suggestion from the meeting, simply copy the `--time` argument used [here](https://github.com/climate-machine/CLIMA/blob/tb/held-suarez-speedup/slurmci-test.toml#L51) to the arguments for Held Suarez.

bischtob commented:
bors r+

bischtob commented:
bors r+

bischtob commented:
bors r-

bischtob commented:
bors r+

bischtob commented:
bors r+
